File: backend/app/schemas/billing.py
from datetime import date, datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field


PlanCode = Literal["free", "basic", "professional"]
PaymentProvider = Literal["mock", "wechat_pay", "alipay", "stripe"]


class PlanOut(BaseModel):
    code: PlanCode
    name: str
    price_cents: int
    customer_limit: int
    ai_daily_limit: int = 0
    campaign_daily_limit: int = 0
    has_export: bool = False


class SubscriptionOut(BaseModel):
    store_id: str
    plan_name: str
    plan_display_name: Optional[str] = None
    status: str
    customer_limit: int = 1000
    ai_used_today: int = 0
    ai_daily_limit: int = 0
    campaign_used_today: int = 0
    campaign_daily_limit: int = 0
    has_export: bool = False
    next_billing_date: Optional[date] = None
    payment_method: Optional[str] = None
    is_active: bool


class CreateOrderRequest(BaseModel):
    plan_name: PlanCode
    provider: PaymentProvider = "mock"


class PaymentOrderOut(BaseModel):
    id: str
    store_id: str
    plan_name: str
    provider: str
    status: str
    amount_cents: int
    currency: str
    checkout_url: Optional[str] = None
    provider_order_id: Optional[str] = None
    paid_at: Optional[datetime] = None
    expires_at: datetime
    created_at: datetime

    model_config = {"from_attributes": True}


class AdminPaymentSummaryOut(BaseModel):
    today_revenue_cents: int
    month_revenue_cents: int
    paid_orders: int
    pending_orders: int
    failed_orders: int
    plan_counts: dict[str, int]
    status_counts: dict[str, int]


class AdminSubscriptionUpdate(BaseModel):
    plan_name: Optional[PlanCode] = None
    status: Optional[str] = None
    customer_limit: Optional[int] = Field(default=None, ge=0)
    next_billing_date: Optional[date] = None

File: frontend/src/api/billing.ts
import http from './http'

export interface Plan {
  code: 'free' | 'basic' | 'professional'
  name: string
  priceCents: number
  customerLimit: number
  aiDailyLimit: number
  campaignDailyLimit: number
  hasExport: boolean
}

export interface Subscription {
  storeId: string
  planName: string
  planDisplayName: string | null
  status: string
  customerLimit: number
  aiUsedToday: number
  aiDailyLimit: number
  campaignUsedToday: number
  campaignDailyLimit: number
  hasExport: boolean
  nextBillingDate: string | null
  paymentMethod: string | null
  isActive: boolean
}

export interface PaymentOrder {
  id: string
  storeId: string
  storeName?: string | null
  planName: string
  provider: string
  status: string
  amountCents: number
  currency: string
  checkoutUrl: string | null
  providerOrderId: string | null
  paidAt: string | null
  expiresAt: string
  createdAt: string
}

export interface AdminPaymentSummary {
  todayRevenueCents: number
  monthRevenueCents: number
  paidOrders: number
  pendingOrders: number
  failedOrders: number
  planCounts: Record<string, number>
  statusCounts: Record<string, number>
}

export interface AdminSubscriptionUpdate {
  planName?: 'free' | 'basic' | 'professional'
  status?: 'active' | 'overdue' | 'cancelled'
  customerLimit?: number
  nextBillingDate?: string | null
}

export const getPlans = () =>
  http.get<Plan[]>('/billing/plans')

export const getSubscription = () =>
  http.get<Subscription>('/billing/subscription')

export const createOrder = (planName: string, provider = 'mock') =>
  http.post<PaymentOrder>('/billing/orders', { planName, provider })

export const getOrder = (orderId: string) =>
  http.get<PaymentOrder>(`/billing/orders/${orderId}`)

export const confirmAlipayReturn = (orderId: string, params: Record<string, string>) =>
  http.post<PaymentOrder>(`/payment/alipay/return/${orderId}`, params)

export const syncAlipayOrder = (orderId: string) =>
  http.post<PaymentOrder>(`/payment/alipay/sync/${orderId}`)

export const mockPay = (orderId: string) =>
  http.post<PaymentOrder>(`/billing/orders/${orderId}/mock-pay`)

export const listAdminOrders = (params?: { page?: number; pageSize?: number; status?: string }) =>
  http.get<{ items: PaymentOrder[]; total: number; page: number; pageSize: number }>('/admin/payment-orders', { params })

export const getAdminPaymentSummary = () =>
  http.get<AdminPaymentSummary>('/admin/payment-summary')

export const listStoreOrders = (storeId: string, params?: { page?: number; pageSize?: number }) =>
  http.get<{ items: PaymentOrder[]; total: number; page: number; pageSize: number }>(`/admin/stores/${storeId}/payment-orders`, { params })

export const updateStoreSubscription = (storeId: string, data: AdminSubscriptionUpdate) =>
  http.put<Subscription>(`/admin/stores/${storeId}/subscription`, data)
